libsolidity/interface/CompilerStack.cpp

```diff
@@ -1921,3 +1949,13 @@ Json::Value CompilerStack::gasEstimates(std::string const& _contractName) const
 
 	return output;
 }
+
+bool CompilerStack::isAnySourceExperimentalSolidity() const
+{
+	return
+		!m_sourceOrder.empty() &&
+		// TODO: Does it make sense to check for any source or just the front?
```

Comment:
Either is fine, since the `analysis()` stage will stop if even one of the source units doesn't have the pragma enabled (this check is performed in `resolveImports()`, right at the beginning of `analysis()`). Basically, checking the front vs. using `ranges::any_of` will be equivalent, so I'd say just check the `front()`.

Reply:
`any_of` is literally going to be the same as the `front()` check; if we want an assert, it would make sense to assert that *all* of the sources have experimental Solidity enabled. Also, what do you mean by "an assumption about something done far away"?

Reply:
True, I said `any_of()`, but I was really thinking about `all_of()` :)

> Also, what do you mean by "an assumption about something done far away"?

That checking one file and ignoring the rest is only fine as long as the assumption that either all files are marked experimental or none holds. And that assumption is enforced by `resolveImports()`, which is 1.5k+ lines away. Which is far enough that someone modifying one of these functions is likely not to notice that the other needs to be changed to match too.